app/command.go

```diff
@@ -202,98 +223,112 @@ func (a *App) ExecuteCommand(args *model.CommandArgs) (*model.CommandResponse, *
+	p.Set("channel_name", channel.Name)
 
-			p.Set("trigger_id", triggerId)
+	p.Set("user_id", args.UserId)
+	p.Set("user_name", user.Username)
 
-			hook, appErr := a.CreateCommandWebhook(cmd.Id, args)
-			if appErr != nil {
-				return nil, model.NewAppError("command", "api.command.execute_command.failed.app_error", map[string]interface{}{"Trigger": trigger}, appErr.Error(), http.StatusInternalServerError)
-			}
-			p.Set("response_url", args.SiteURL+"/hooks/commands/"+hook.Id)
-
-			var req *http.Request
-			if cmd.Method == model.COMMAND_METHOD_GET {
-				req, _ = http.NewRequest(http.MethodGet, cmd.URL, nil)
-
-				if req.URL.RawQuery != "" {
-					req.URL.RawQuery += "&"
-				}
-				req.URL.RawQuery += p.Encode()
-			} else {
-				req, _ = http.NewRequest(http.MethodPost, cmd.URL, strings.NewReader(p.Encode()))
-			}
+	p.Set("command", "/"+trigger)
+	p.Set("text", message)
 
-			req.Header.Set("Accept", "application/json")
-			req.Header.Set("Authorization", "Token "+cmd.Token)
-			if cmd.Method == model.COMMAND_METHOD_POST {
-				req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-			}
+	p.Set("trigger_id", args.TriggerId)
 
-			resp, err := a.HTTPService.MakeClient(false).Do(req)
-			if err != nil {
-				return nil, model.NewAppError("command", "api.command.execute_command.failed.app_error", map[string]interface{}{"Trigger": trigger}, err.Error(), http.StatusInternalServerError)
-			}
-			if resp.StatusCode != http.StatusOK {
-				defer resp.Body.Close()
-				body, _ := ioutil.ReadAll(resp.Body)
-				return nil, model.NewAppError("command", "api.command.execute_command.failed_resp.app_error", map[string]interface{}{"Trigger": trigger, "Status": resp.Status}, string(body), http.StatusInternalServerError)
-			}
+	hook, appErr := a.CreateCommandWebhook(cmd.Id, args)
+	if appErr != nil {
+		return cmd, nil, model.NewAppError("command", "api.command.execute_command.failed.app_error", map[string]interface{}{"Trigger": trigger}, appErr.Error(), http.StatusInternalServerError)
+	}
+	p.Set("response_url", args.SiteURL+"/hooks/commands/"+hook.Id)
 
-			response, err := model.CommandResponseFromHTTPBody(resp.Header.Get("Content-Type"), resp.Body)
-			if err != nil {
-				return nil, model.NewAppError("command", "api.command.execute_command.failed.app_error", map[string]interface{}{"Trigger": trigger}, err.Error(), http.StatusInternalServerError)
-			}
-			if response == nil {
-				return nil, model.NewAppError("command", "api.command.execute_command.failed_empty.app_error", map[string]interface{}{"Trigger": trigger}, "", http.StatusInternalServerError)
-			}
+	return a.doCommandRequest(cmd, p)
+}
 
-			response.TriggerId = clientTriggerId
+func (a *App) doCommandRequest(cmd *model.Command, p url.Values) (*model.Command, *model.CommandResponse, *model.AppError) {
+	// Prepare the request
+	var req *http.Request
+	if cmd.Method == model.COMMAND_METHOD_GET {
+		req, _ = http.NewRequest(http.MethodGet, cmd.URL, nil)
```

Comment:
I _just_ fixed a bug on this. We must not ignore error codes returned from standard library.

Reply:
Yep, I'll add that in